### ISSUE TITLE
Squiz.Commenting.FunctionComment.IncorrectTypeHint don't recognize namespace aliases

### ISSUE THREAD

**Issue by FrittenKeeZ** [squizlabs/PHP_CodeSniffer#3651]
**Describe the bug**
Using namespace alises (imports) the rule thinks a wrong type hint is used.

**Code sample**
```php
<?php

namespace App\Policies;

use Illuminate\Auth\Access\HandlesAuthorization;
use Illuminate\Contracts\Auth\Authenticatable as User;

class CustomerPolicy
{
    use HandlesAuthorization;

    /**
     * Determine whether the user can view any models.
     *
     * @param \Illuminate\Contracts\Auth\Authenticatable $user
     *
     * @return mixed
     */
    public function viewAny(User $user)
    {
        return true;
    }
```

**Custom ruleset**
```xml
<rule ref="PSR12">
    <exclude name="Squiz.Commenting.FunctionComment.IncorrectParamVarName" />
    <exclude name="Squiz.Commenting.FunctionComment.InvalidReturn" />
    <exclude name="Squiz.Commenting.FunctionComment.MissingParamComment" />
    <exclude name="PEAR.Commenting.FunctionComment.MissingParamComment" />
</rule>
<rule ref="Squiz.Commenting.FunctionComment.TypeHintMissing">
    <exclude-pattern>*/app/Http/Middleware/*</exclude-pattern>
    <exclude-pattern>*/app/Http/Resources/*</exclude-pattern>
    <exclude-pattern>*/app/Nova/*</exclude-pattern>
</rule>
```

**To reproduce**
1. In a Laravel 9 project
1. Create a file called `app\Policies\CustomerPolicy.php` with the code sample above...
2. Run `phpcs app\Policies\CustomerPolicy.php ...`
3. See error message displayed
```
Expected type hint "\Illuminate\Contracts\Auth\Authenticatable"; found "User" for $user (Squiz.Commenting.FunctionComment.IncorrectTypeHint)
```

**Expected behavior**
The aliased namespace should correctly be detected and no error given.

**Versions (please complete the following information):**
 - OS: 12.5
 - PHP: 8.0
 - PHPCS: .9.5
 - Standard: PSR12

**Comment by mreduar** [comment]
Hello, have you resolved this issue?

**Comment by FrittenKeeZ** [comment]
@mreduar No, it never got resolved, so we ended up switching to Laravel Pint.

**Comment by rodrigoprimo** [comment]
I believe this is a duplicate of https://github.com/squizlabs/PHP_CodeSniffer/issues/325. Linking for visibility.

**Comment by mreduar** [comment]
> I believe this is a duplicate of [#325](https://github.com/squizlabs/PHP_CodeSniffer/issues/325). Linking for visibility.

Yes, that's correct. The workaround I implemented was to remove the FQCN from the PHP Doc Block and, in its place, use the alias of the imported class.
